adele-booking/src/main/java/com/microservicesteam/adele/booking/domain/BookingService.java

```diff
@@ -0,0 +1,94 @@
+package com.microservicesteam.adele.booking.domain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.microservicesteam.adele.booking.boundary.web.EventPublisher;
+import com.microservicesteam.adele.messaging.EventBasedService;
+import com.microservicesteam.adele.ticketmaster.commands.BookTickets;
+import com.microservicesteam.adele.ticketmaster.events.TicketsBooked;
+import com.microservicesteam.adele.ticketmaster.events.TicketsCancelled;
+import com.microservicesteam.adele.ticketmaster.events.TicketsCreated;
+import com.microservicesteam.adele.ticketmaster.model.BookedTicket;
+import com.microservicesteam.adele.ticketmaster.model.FreeTicket;
+import com.microservicesteam.adele.ticketmaster.model.Position;
+import com.microservicesteam.adele.ticketmaster.model.Ticket;
+
+@Service
+public class BookingService extends EventBasedService {
+
+    private final BookingIdGenerator bookingIdGenerator;
+    private final EventPublisher eventPublisher;
+    Map<Position, Ticket> ticketRepository;
+
+
+    BookingService(EventBus eventBus, BookingIdGenerator bookingIdGenerator, EventPublisher eventPublisher) {
+        super(eventBus);
+        this.bookingIdGenerator = bookingIdGenerator;
+        this.eventPublisher = eventPublisher;
+        ticketRepository = new HashMap<>();
+    }
+
+    public List<Ticket> getTicketsStatus() {
+        return ImmutableList.copyOf(ticketRepository.values());
+    }
+
+    public BookingResponse bookTickets(BookingRequest bookingRequest) {
+        String bookingId = bookingIdGenerator.generateBookingId();
+
+        List<Position> requestedPositions = toPositions(bookingRequest);
+
+        eventBus.post(BookTickets.builder()
+                .bookingId(bookingId)
+                .addAllPositions(requestedPositions)
+                .build());
+
+        return BookingResponse.builder()
+                .withBookingId(bookingId)
+                .build();
+    }
+
+    @Subscribe
+    public void handleEvent(TicketsCreated ticketsCreated) {
+        ticketsCreated.positions()
+                .forEach(position -> ticketRepository.put(position, FreeTicket.builder()
+                        .position(position)
+                        .build()));
+        eventPublisher.publish(ticketsCreated);
```

Comment:
I wasn't sure either 😄  Yeah, it is simpler to forgot about cases like that. I will delete notifying UI about Tickets created.